### PR TITLE
Updates for Gaea C5 & F2/F5 Transition + Fix out of bound errors in block_atmos_copy routines in fv3atm #2124

### LINKED ISSUE #2096: Bring in detect_machine.sh from global workflow for consistency across the community.
## Description

The global workflow team has an updated script to detect the machine the script is currently using. It uses a regexp method to detect the machine either by hostname or the directory structure. This script should be kept consistent across the ufs community.

## Solution

* Bring in (update): https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/detect_machine.sh to the ufs-weather-model 
* Add `global-workflow admins` to the codeowners for detect_machine.sh so they are aware of any changes we make to the script and can bring them into their repository.

### PR DESCRIPTION
**Pull request by ulmononian**
## Commit Queue Requirements:

- [x] Fill out all sections of this template.
- [x] All sub component pull requests have been reviewed by their code managers.
- [x] Run the full RT suite (compared to current baselines) on either Hera/Derecho/Hercules AND have committed the log to my PR branch.
- [ ] Add list of all failed regression tests in "Regression Tests" section.

### Description
spack-stack locations have been changed to accommodate the Gaea C5 F2 to F5 transition (see #2101 and https://github.com/ufs-community/ufs-weather-model/issues/2101#issuecomment-1906521268). This PR updates the WM to accommodate these changes, as well as changes the `Gaea C5` name to simply `Gaea`.

### Commit Message

- Gaea C5 modulefile & DISKNM update: closes issues #2101
- Bring in the global-workflow detect_machine.sh to keep consistent between projects. (Closes #2096 ) 
- Fix out of bound errors in block_atmos_copy routines in fv3tm

### Priority
- [ ] Critical Bugfix (This PR contains a critical bug fix and should be prioritized.)
- [x] High (This PR contains a feature or fix needed for a time-sensitive project (eg, retrospectives, implementations))
- [ ] Normal

### Blocking Dependencies
https://github.com/NOAA-EMC/fv3atm/pull/778

### Git Issues Fixed By This PR
Partially addresses #2101 
Closes https://github.com/ufs-community/ufs-weather-model/pull/2124
Closes https://github.com/ufs-community/ufs-weather-model/pull/2111

### Subcomponent (with links)
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [x] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [x] none

### Input data
- [x] No changes are expected to input data.
- [ ] Changes are expected to input data:
  - [ ] New input data.
  - [ ] Updated input data.

### Regression Tests:
- [x] No changes are expected to any regression test.
- [ ] Changes are expected to the following tests:
<details><summary>FAILED REGRESSION TESTS</summary>

</details>

### Libraries

- [x] Not Needed
- [ ] Needed
  - [ ] Create separate issue in [JCSDA/spack-stack](https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
  - [ ] Add issue link from JCSDA/spack-stack following this item 

### Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- CI
  - [x] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment

### CODE REVIEW

**Comment by ulmononian** [comment]
@RatkoVasic-NOAA @natalie-perlin @jkbk2004 @zach1221 FYI

**Comment by BrianCurtis-NOAA** [comment]
Why is it breaking Rocoto? We do need the name to go from gaea-c5 to gaea.

**Comment by ulmononian** [comment]
> Why is it breaking Rocoto? We do need the name to go from gaea-c5 to gaea.

not exactly certain why, but with the name changed in all necessary places, when you submit RTs w/ `-r`, rocoto complains about the partition (whether it is set to `batch` or `c5`). note that, although we set `--partition=batch` in `fv3_slurm.IN_gaea-c5`, rocoto was overwriting this setting and taking the `PARTITION` value from `rt.sh`(tested by changing `PARTITION` to `batch` in `rt.sh`). the output looked like:

```
01/26/24 13:22:36 EST :: rocoto_workflow.xml :: WARNING: job submission failed: sbatch: error: invalid partition specified: batch
sbatch: error: Batch job submission failed: Invalid partition name specified

01/26/24 12:09:18 EST :: rocoto_workflow.xml :: WARNING: job submission failed: sbatch: error: invalid partition specified: c5
sbatch: error: Batch job submission failed: Invalid partition name specified
```

with the name set back to `gaea-c5` in all places, rocoto runs just fine and does not complain about the partition (which is again properly set from fv3_slurm.IN_gaea-c5).

**Comment by BrianCurtis-NOAA** [comment]
I think you'll want to edit: https://github.com/ulmononian/ufs-weather-model/blob/3db68113d2a92e1b59f2fe34cdac3caaec1774fb/tests/rt.sh#L661-L664 to match the partition names rocoto uses

**Comment by zach1221** [comment]
@ulmononian I can reach back out to the gaea admins to see why the name change is causing that issue with Rocoto. It'll have to be resolved for Brian's PR.

> I think you'll want to edit: https://github.com/ulmononian/ufs-weather-model/blob/3db68113d2a92e1b59f2fe34cdac3caaec1774fb/tests/rt.sh#L661-L664 to match the partition names rocoto uses

I can test this. I thought we had updated this section in the original PR.

**Comment by ulmononian** [comment]
> I think you'll want to edit: https://github.com/ulmononian/ufs-weather-model/blob/3db68113d2a92e1b59f2fe34cdac3caaec1774fb/tests/rt.sh#L661-L664 to match the partition names rocoto uses

not sure i know what you mean...doesn't rocoto get the partition name from fv3_slurm.IN_<machine>?

**Comment by ulmononian** [comment]
> @ulmononian I can reach back out to the gaea admins to see why the name change is causing that issue with Rocoto. It'll have to be resolved for Brian's PR.
> 
> > I think you'll want to edit: https://github.com/ulmononian/ufs-weather-model/blob/3db68113d2a92e1b59f2fe34cdac3caaec1774fb/tests/rt.sh#L661-L664 to match the partition names rocoto uses
> 
> I can test this. I thought we had updated this section in the original PR.

we did update this section (your PR changed gaea-c5 to gaea, here). but i made all of these name changes, working from develop, and rocoto started to break when the name changes were added.

**Comment by ulmononian** [comment]
@jkbk2004 @zach1221 @BrianCurtis-NOAA 

fyi: control_c48 and cpld_control_p8 pass (new baseline created for testing) for this PR.

**Comment by BrianCurtis-NOAA** [comment]
> > I think you'll want to edit: https://github.com/ulmononian/ufs-weather-model/blob/3db68113d2a92e1b59f2fe34cdac3caaec1774fb/tests/rt.sh#L661-L664 to match the partition names rocoto uses
> 
> not sure i know what you mean...doesn't rocoto get the partition name from fv3_slurm.IN_?

I don't deal with ROCOTO much, but if I recall correctly UFSWM had to accommodate ROCOTO having different partition names then the main cluster at one point and is why that section of code exists still in rt.sh. the fv3_slurm..in file is not specific to rocoto or ecflow.

**Comment by RatkoVasic-NOAA** [comment]
@ulmononian @BrianCurtis-NOAA just checked. There must be something else that made rocoto fail. Here is directory with necessary changes for running with rocoto and ecflow:
```
Gaea:/gpfs/f5/epic/scratch/Ratko.Vasic/2115/ufs-weather-model (current develop branch + changes)
Gaea:/gpfs/f5/epic/scratch/Ratko.Vasic/2115/ufs-weather-model>git status
Changes not staged for commit:
        deleted:    cmake/configure_gaea-c5.intel.cmake
        modified:   doc/UsersGuide/source/BuildingAndRunning.rst
        deleted:    modulefiles/ufs_gaea-c5.intel.lua
        modified:   tests/auto-jenkins/start_rt_auto_jenkins.sh
        modified:   tests/auto/start_rt_auto.sh
        modified:   tests/compile.sh
        modified:   tests/default_vars.sh
        modified:   tests/detect_machine.sh
        deleted:    tests/fv3_conf/compile_slurm.IN_gaea-c5
        deleted:    tests/fv3_conf/fv3_slurm.IN_gaea-c5
        deleted:    tests/logs/RegressionTests_gaea-c5.log
        modified:   tests/module-setup.sh
        modified:   tests/rt.sh
        modified:   tests/rt_utils.sh
        modified:   tests/run_test.sh
        modified:   tests/tests/regional_atmaq
        modified:   tests/tests/regional_atmaq_debug
        modified:   tests/tests/regional_atmaq_faster

Untracked files:
        cmake/configure_gaea.intel.cmake
        modulefiles/ufs_gaea.intel.lua
        tests/fv3_conf/compile_slurm.IN_gaea
        tests/fv3_conf/fv3_slurm.IN_gaea
        tests/logs/RegressionTests_gaea.log

```

**Comment by RatkoVasic-NOAA** [comment]
**NOTE**
two files:
```
./tests/auto/start_rt_auto.sh
./tests/auto-jenkins/start_rt_auto_jenkins.sh

```
They need new path to `miniconda3/4.8.3/envs/ufs-weather-model `which I couldn't find on Gaea-C5

**Comment by zach1221** [comment]
@RatkoVasic-NOAA thanks for passing this along. I'll test the changes in my own directory against rt.conf using both rocoto/ecflow.

**Comment by ulmononian** [comment]
thanks to @RatkoVasic-NOAA, the rocoto issue has been resolved. full intel RT suite running here: `/gpfs/f5/epic/scratch/Cameron.Book/FV3_RT/rt_141949` (base dir: `/gpfs/f5/epic/scratch/Cameron.Book/fix_gaeaPR/tests`).

**Comment by ulmononian** [comment]
@jkbk2004 @BrianCurtis-NOAA @zach1221 all intel RTs pass for this PR. RegressionTests_gaea.log was uploaded to `/tests/logs` dir..

**Comment by ulmononian** [comment]
full intel RT log: [RegressionTests_gaea(2).log](https://github.com/ufs-community/ufs-weather-model/files/14114859/RegressionTests_gaea.2.log)

File: tests/detect_machine.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ case $(hostname -f) in
   gaea52.ncrc.gov)         MACHINE_ID=gaea-c5 ;; ### gaea52
   gaea53.ncrc.gov)         MACHINE_ID=gaea-c5 ;; ### gaea53
   gaea54.ncrc.gov)         MACHINE_ID=gaea-c5 ;; ### gaea54
-  gaea55.ncrc.gov)         MACHINE_ID=gaea-c5 ;; ### gaea55
+  gaea55.ncrc.gov)         MACHINE_ID=gaea ;; ### gaea55
   gaea56.ncrc.gov)         MACHINE_ID=gaea-c5 ;; ### gaea56
   gaea57.ncrc.gov)         MACHINE_ID=gaea-c5 ;; ### gaea57
   gaea58.ncrc.gov)         MACHINE_ID=gaea-c5 ;; ### gaea58
```
Comment (BrianCurtis-NOAA) [inline tests/detect_machine.sh]: You will want to rename all of these MACHINE_ID to just "gaea"

**Comment by BrianCurtis-NOAA** [comment]
Please don't remove all the GNU tests.from rt.conf

File: tests/detect_machine.sh (hunk 1 above)
Comment (ulmononian) [reply tests/detect_machine.sh]: @BrianCurtis-NOAA it looks to me like they are all changed to `gaea` only...am i missing something?

**Comment by ulmononian** [comment]
> Please don't remove all the GNU tests.from rt.conf

i restored the develop version of `rt.conf` -- sorry about that.

**Comment by ulmononian** [comment]
> Please don't remove all the GNU tests.from rt.conf

@BrianCurtis-NOAA i restored the develop version of rt.conf -- sorry about that.

**Comment by zach1221** [comment]
@ulmononian you used rocoto to run your tests, correct? I'm trying to run through rt.conf using ecflow to create baselines, and the tests show as having completed in the "out" logs here: /gpfs/f5/epic/scratch/Zachary.Shrader/FV3_RT/rt_210443

But the RegressionTest directory remains blank when trying to create baselines: /gpfs/f5/epic/scratch/Zachary.Shrader/FV3_RT/REGRESSION_TEST

My working directory test logs also show a failure, example :/gpfs/f5/epic/scratch/Zachary.Shrader/rt-2115/tests/logs/log_gaea/run_130_atmaero_control_p8_intel.log (ecflow_client: /opt/cray/pe/gcc/10.3.0/snos/lib64/libstdc++.so.6: version `CXXABI_1.3.13' not found (required by ecflow_client))

Can you try creating baselines with ecflow again, and let me know if you experience this behavior? ./rt.sh -a nggps_emc -c -e -l rt.conf

**Comment by ulmononian** [comment]
> @ulmononian you used rocoto to run your tests, correct? I'm trying to run through rt.conf using ecflow to create baselines, and the tests show as having completed in the "out" logs here: /gpfs/f5/epic/scratch/Zachary.Shrader/FV3_RT/rt_210443
> 
> But the RegressionTest directory remains blank when trying to create baselines: /gpfs/f5/epic/scratch/Zachary.Shrader/FV3_RT/REGRESSION_TEST
> 
> My working directory test logs also show a failure, example :/gpfs/f5/epic/scratch/Zachary.Shrader/rt-2115/tests/logs/log_gaea/run_130_atmaero_control_p8_intel.log (ecflow_client: /opt/cray/pe/gcc/10.3.0/snos/lib64/libstdc++.so.6: version `CXXABI_1.3.13' not found (required by ecflow_client))
> 
> Can you try creating baselines with ecflow again, and let me know if you experience this behavior? ./rt.sh -a nggps_emc -c -e -l rt.conf

i didn't create baselines w/ ecflow, but @RatkoVasic-NOAA tried i believe.

@RatkoVasic-NOAA @zach1221 btw: my full intel RT (run w/ rocoto) is here if you want to check baselines: `/gpfs/f5/epic/scratch/Cameron.Book/FV3_RT/rt_141949`

**Comment by jkbk2004** [comment]
> @ulmononian you used rocoto to run your tests, correct? I'm trying to run through rt.conf using ecflow to create baselines, and the tests show as having completed in the "out" logs here: /gpfs/f5/epic/scratch/Zachary.Shrader/FV3_RT/rt_210443
> 
> But the RegressionTest directory remains blank when trying to create baselines: /gpfs/f5/epic/scratch/Zachary.Shrader/FV3_RT/REGRESSION_TEST
> 
> My working directory test logs also show a failure, example :/gpfs/f5/epic/scratch/Zachary.Shrader/rt-2115/tests/logs/log_gaea/run_130_atmaero_control_p8_intel.log (ecflow_client: /opt/cray/pe/gcc/10.3.0/snos/lib64/libstdc++.so.6: version `CXXABI_1.3.13' not found (required by ecflow_client))
> 
> Can you try creating baselines with ecflow again, and let me know if you experience this behavior? ./rt.sh -a nggps_emc -c -e -l rt.conf

@ulmononian @RatkoVasic-NOAA I see same error with ecflow.

**Comment by jkbk2004** [comment]
rocoto works ok.

**Comment by zach1221** [comment]
> rocoto works ok.

Yes rocoto works great.

**Comment by RatkoVasic-NOAA** [comment]
@zach1221 @jkbk2004 do you still see the problems with ecFlow on Gaea? I have 2 directories where I tested ecFlow and rocoto. I started from current develop branch.
Here are changes just for ecFlow:
`/gpfs/f5/epic/scratch/Ratko.Vasic/ufs-weather-model`
And here included changes for ecFlow, rocoto and renamed gaea-c5 to gaea:
`/gpfs/f5/epic/scratch/Ratko.Vasic/2115/ufs-weather-model`

**Comment by zach1221** [comment]
> /gpfs/f5/epic/scratch

Hi, @RatkoVasic-NOAA . I did still have the issue when I tested with ecflow last night, but I can try again now. I'll keep you posted on how it goes this time.

**Comment by zach1221** [comment]
@RatkoVasic-NOAA I copied your weather-model repo from /gpfs/f5/epic/scratch/Ratko.Vasic/2115/ufs-weather-model. Still having the failures here. I used epic account this time.
gpfs/f5/epic/scratch/Zachary.Shrader/ufs-weather-model/tests/logs/log_gaea

**Comment by RatkoVasic-NOAA** [comment]
> @RatkoVasic-NOAA I copied your weather-model repo from /gpfs/f5/epic/scratch/Ratko.Vasic/2115/ufs-weather-model. Still having the failures here. I used epic account this time. gpfs/f5/epic/scratch/Zachary.Shrader/ufs-weather-model/tests/logs/log_gaea

@zach1221 can you please change permissions to your /gpfs/f5/epic/scratch/Zachary.Shrader/ directory.

**Comment by zach1221** [comment]
> > @RatkoVasic-NOAA I copied your weather-model repo from /gpfs/f5/epic/scratch/Ratko.Vasic/2115/ufs-weather-model. Still having the failures here. I used epic account this time. gpfs/f5/epic/scratch/Zachary.Shrader/ufs-weather-model/tests/logs/log_gaea
> 
> @zach1221 can you please change permissions to your /gpfs/f5/epic/scratch/Zachary.Shrader/ directory.

@RatkoVasic-NOAA sure, this should be done.

**Comment by RatkoVasic-NOAA** [comment]
@jkbk2004 @zach1221 we (@ulmononian and I) found solution for ecFlow problem. I made changes to this PR. Can you please check if it is working now.

**Comment by jkbk2004** [comment]
> @jkbk2004 @zach1221 we (@ulmononian and I) found solution for ecFlow problem. I made changes to this PR. Can you please check if it is working now.

@RatkoVasic-NOAA Great! I am testing now. Looks like jobs running with ecflow. I will keep posting.

**Comment by zach1221** [comment]
PR 2028 has been merged. @ulmononian I think we're looking to begin testing on this PR next, could you sync up your branch, please? Additionally, we had been considering combining this PR with @BrianCurtis-NOAA's PR #2111 . If you're ok with this, Cameron, could you combine in Brian's changes as well? Fyi @jkbk2004

**Comment by caihong-ocean** [comment]
I could not access /gpfs/f5/epic/scratch/role.epic because of permission denied.  The log message is
"source /gpfs/f5/epic/scratch/role.epic/contrib/Lmod_init_C5.sh"
"module-setup.sh: line 84: /gpfs/f5/epic/scratch/role.epic/contrib/Lmod_init_C5.sh: Permission denied"

**Comment by RatkoVasic-NOAA** [comment]
> I could not access /gpfs/f5/epic/scratch/role.epic because of permission denied. The log message is "source /gpfs/f5/epic/scratch/role.epic/contrib/Lmod_init_C5.sh" "module-setup.sh: line 84: /gpfs/f5/epic/scratch/role.epic/contrib/Lmod_init_C5.sh: Permission denied"

@caihong-ocean I changed permission.

**NOTE** I already changed permission before (`/gpfs/f5/epic/scratch/role.epic`) , but somehow all subdirectories have permissions back to user-read-only:
```
drwx------  2 Alexander.Richert          epic 4096 Jan  5 21:11 Alexander.Richert/
drwx------  4 Anil.Kumar                 epic 4096 Feb  2 11:09 Anil.Kumar/
drwx------  2 Anna.Kimball               epic 4096 Jan  5 21:11 Anna.Kimball/
drwx------  2 Brian.Curtis               epic 4096 Jan  5 21:11 Brian.Curtis/
drwx------  7 Cameron.Book               epic 4096 Feb  5 20:14 Cameron.Book/
drwx------  2 Chandin.Wilson             epic 4096 Jan  5 21:11 Chandin.Wilson/
drwx------  2 Christina.Holt             epic 4096 Jan  5 21:11 Christina.Holt/
drwx------  2 David.Burrows              epic 4096 Feb  2 13:52 David.Burrows/
drwx------  2 David.Huber                epic 4096 Jan  5 21:11 David.Huber/
drwx------  2 Dom.Heinzeller             epic 4096 Jan  5 21:11 Dom.Heinzeller/
drwx------  2 Edward.Snyder              epic 4096 Jan  5 21:11 Edward.Snyder/
drwx------  2 Emily.L.Carpenter          epic 4096 Jan  5 21:11 Emily.L.Carpenter/
drwx------  3 Fernando.Andrade-maldonado epic 4096 Jan 18 15:34 Fernando.Andrade-maldonado/
drwx------  2 Gillian.Petro              epic 4096 Jan  5 21:11 Gillian.Petro/
drwx------  2 Henrique.Alves             epic 4096 Jan  5 21:11 Henrique.Alves/
drwx------  5 Jong.Kim                   epic 4096 Feb  2 10:15 Jong.Kim/
drwx------  2 Keven.Blackman             epic 4096 Jan  5 21:11 Keven.Blackman/
drwx------  2 Krishna.Kumar              epic 4096 Jan  5 21:11 Krishna.Kumar/
drwx------  2 Kristopher.Booker          epic 4096 Jan  5 21:11 Kristopher.Booker/
drwx------  2 Mark.Potts                 epic 4096 Jan  5 21:11 Mark.Potts/
drwx------  3 Michael.Lueken             epic 4096 Jan 31 11:19 Michael.Lueken/
drwx------  3 Natalie.Perlin             epic 4096 Feb  1 11:13 Natalie.Perlin/
drwx------  2 Naureen.Bharwani           epic 4096 Jan  5 21:11 Naureen.Bharwani/
drwx------  2 Paul.Madden                epic 4096 Jan  5 21:11 Paul.Madden/
drwx------  8 Ratko.Vasic                epic 4096 Feb  5 12:42 Ratko.Vasic/
drwx------  2 Rhaesung.Kim               epic 4096 Jan  5 21:11 Rhaesung.Kim/
drwx------  5 role.epic                  epic 4096 Feb  5 18:37 role.epic/
drwx------  2 Steven.Bongiovanni         epic 4096 Jan  5 21:11 Steven.Bongiovanni/
drwx------  5 Zachary.Shrader            epic 4096 Feb  1 09:48 Zachary.Shrader/
```
I'm afraid that Gaea system have somewhere cron reverting permissions. If this happens again, we have to see with Gaea administrators.

**Comment by jkbk2004** [comment]
@RatkoVasic-NOAA @ulmononian @caihong-ocean what about somewhere putting the epic contrib space in /gpfs/f5/epic/world-shared ? I think intention of world-shared space is to let all others access.

**Comment by RatkoVasic-NOAA** [comment]
> @RatkoVasic-NOAA @ulmononian @caihong-ocean what about somewhere putting the epic contrib space in /gpfs/f5/epic/world-shared ? I think intention of world-shared space is to let all others access.

@jkbk2004 I copied Lmod_init_C5.sh to `/gpfs/f5/epic/world-shared/contrib/Lmod_init_C5.sh`
We have to find all places where old path is used to change it to new place (SRW, WM,...)

**Comment by BrianCurtis-NOAA** [comment]
@ulmononian Were you having trouble adding my #2111 to this?

**Comment by jkbk2004** [comment]
> @ulmononian Were you having trouble adding my #2111 to this?

@BrianCurtis-NOAA Due to syncing issue (hashes), Git merge wasn't smooth to pick up correct FV3/CDEPS/CMEPS hashes. Looks like it happens with other branches on some other PRs. I manually synced up your branch. @ulmononian I think it should be ok now. can you try to combine in #2111 ?

**Comment by caihong-ocean** [comment]
Similar permission issue also occurred for 
/ncrc/proj/epic/rocoto/modulefiles
      The scripts has trouble to executed the following code in rt.sh
"module use /ncrc/proj/epic/rocoto/modulefiles
  module load rocoto"
        Looks like all the shared libraries/codes need to put in project shared directories. I could access  /ncrc/proj/epic yesterday , but failed today.

**Comment by ulmononian** [comment]
@jkbk2004 @BrianCurtis-NOAA pulled in #2111 again.

**Comment by RatkoVasic-NOAA** [comment]
> Similar permission issue also occurred for /ncrc/proj/epic/rocoto/modulefiles The scripts has trouble to executed the following code in rt.sh "module use /ncrc/proj/epic/rocoto/modulefiles module load rocoto" Looks like all the shared libraries/codes need to put in project shared directories. I could access /ncrc/proj/epic yesterday , but failed today.

@caihong-ocean can you please share your script rt.sh with us? I have no problems accessing /ncrc/proj/epic directory from my personal account. I understand if you couldn't access /gpfs/f5/epic/scratch/role.epic directory.

**Comment by pjpegion** [comment]
The spack-stack module files will also need to move

**Comment by caihong-ocean** [comment]
@RatkoVasic-NOAA Here is the related lines in rt.sh
"
elif [[ $MACHINE_ID = gaea-c5 ]]; then

  module use /ncrc/proj/epic/rocoto/modulefiles
  module load rocoto
  ROCOTORUN=$(which rocotorun)
  ROCOTOSTAT=$(which rocotostat)
  ROCOTOCOMPLETE=$(which rocotocomplete)
  ROCOTO_SCHEDULER=slurm

  module load PrgEnv-intel/8.3.3
  module load intel-classic/2023.1.0
  module load cray-mpich/8.1.25
  module use /ncrc/proj/epic/spack-stack/modulefiles
  module load python/3.9.12
  module use /ncrc/proj/epic/spack-stack/modulefiles
  module use /ncrc/proj/epic/rocoto/modulefiles
  module load ecflow/5.8.4
" 
     I guessed in F5 system,  people are assigned to individual projects. if people have user accounts in project  "epic", they can access the files.

**Comment by RatkoVasic-NOAA** [comment]
I guessed in F5 system, people are assigned to individual projects. if people have user accounts in project "epic", they can access the files.

@caihong-ocean can you please run these commands from your user account so we can see where is the problem:
```
ls -ld /ncrc
ls -ld /ncrc/proj
ls -ld /ncrc/proj/epic
ls -ld /ncrc/proj/epic/rocoto
ls -ld /ncrc/proj/epic/rocoto/modulefiles
ls -ld /ncrc/proj/epic/rocoto/modulefiles/rocoto
```
thanks

**Comment by jkbk2004** [comment]
> I guessed in F5 system, people are assigned to individual projects. if people have user accounts in project "epic", they can access the files.
> 
> @caihong-ocean can you please run these commands from your user account so we can see where is the problem:
> 
> ```
> ls -ld /ncrc
> ls -ld /ncrc/proj
> ls -ld /ncrc/proj/epic
> ls -ld /ncrc/proj/epic/rocoto
> ls -ld /ncrc/proj/epic/rocoto/modulefiles
> ls -ld /ncrc/proj/epic/rocoto/modulefiles/rocoto
> ```
> 
> thanks

If there is boundary btw projects, the permission issue continue.

**Comment by RatkoVasic-NOAA** [comment]
@jkbk2004 first sentence was my quote :-)

**Comment by pjpegion** [comment]
issue is at the epic level:
ls -l 
drwxrwx---  6 root epic           4096 Jan 19 13:58 epic

this directory is only accessible to users who are in the epic group.

**Comment by caihong-ocean** [comment]
@RatkoVasic-NOAA  Here is what I got:
Caihong.Wen@gaea58:~> ls -ld /ncrc
drwxr-xr-x 2 root root 160 Jan 17 12:40 /ncrc
Caihong.Wen@gaea58:~> ls -ld /ncrc/proj
lrwxrwxrwx 1 root root 22 Jan 17 12:40 /ncrc/proj -> /autofs/ncrc-svm1_proj
Caihong.Wen@gaea58:~> ls -ld /ncrc/proj/epic
drwxrwx--- 6 root epic 4096 Jan 19 13:58 /ncrc/proj/epic
Caihong.Wen@gaea58:~> ls -ld /ncrc/proj/epic/rocoto
ls: cannot access '/ncrc/proj/epic/rocoto': Permission denied
Caihong.Wen@gaea58:~> ls -ld /ncrc/proj/epic/rocoto/modulefiles
ls: cannot access '/ncrc/proj/epic/rocoto/modulefiles': Permission denied
Caihong.Wen@gaea58:~> ls -ld /ncrc/proj/epic/rocoto/modulefiles/rocoto
ls: cannot access '/ncrc/proj/epic/rocoto/modulefiles/rocoto': Permission denied

**Comment by jkbk2004** [comment]
> issue is at the epic level: ls -l drwxrwx--- 6 root epic 4096 Jan 19 13:58 epic

sounds like sudo permission is needed at that level. @ulmononian @RatkoVasic-NOAA you may ask admins.

**Comment by RatkoVasic-NOAA** [comment]
@jkbk2004 @caihong-ocean @ulmononian @natalie-perlin 
Oh, I see now. Admins changed permissions:
drwxrwx--- 6 root epic 4096 Jan 19 2024 13:58:24 /ncrc/proj/epic

**Comment by RatkoVasic-NOAA** [comment]
I'm contacting Gaea admins now

**Comment by natalie-perlin** [comment]
> > @RatkoVasic-NOAA @ulmononian @caihong-ocean what about somewhere putting the epic contrib space in /gpfs/f5/epic/world-shared ? I think intention of world-shared space is to let all others access.
> 
> @jkbk2004 I copied Lmod_init_C5.sh to `/gpfs/f5/epic/world-shared/contrib/Lmod_init_C5.sh` We have to find all places where old path is used to change it to new place (SRW, WM,...)

@RatkoVasic-NOAA , @jkbk2004 @ulmononian - No specific file for Lmod initialization is needed, the Lmod_init_C5.sh could be deleted. For Lmod initialization on Gaea in files such as module-setup.sh, or other similar places if any, use "module reset" for Gaea-c5.

Please note that GFDL and Gaea system administration team specifically asked _not_ to use   /gpfs/f5/epic/world-shared for any software installation. This is a parallel filesystem and could be used for data sharing. A specific location has been created to install all of the software under **/ncrc/proj/epic** .

**Comment by jkbk2004** [comment]
@RatkoVasic-NOAA @ulmononian what about wpo space there? I am able to write there. it looks good: [role.epic@gaea58:/ncrc/proj/wpo
```
drwxrwsr-x  2 root wpo            4096 Dec 19  2022 wpo
```

**Comment by natalie-perlin** [comment]
> I'm contacting Gaea admins now

Yes, please!!! You could mention that Gaea admins asked not to install any software under /gpfs/f5/epic/world-shared/, and created a dedicated directory for the software under /ncrc/proj/epic . This still needs to have a read access for all system users!

File: tests/rt.sh [hunk 2]
```diff
@@ -304,30 +304,32 @@ elif [[ $MACHINE_ID = acorn ]]; then
...
   ECF_PORT=$(( $(id -u) + 1500 ))
 
-  DISKNM=/lustre/f2/pdata/ncep/role.epic/C5/RT
+  DISKNM=/gpfs/f5/epic/world-shared/UFS-WM_RT
   QUEUE=normal
   COMPILE_QUEUE=normal
   PARTITION=c5
-  STMP=/lustre/f2/scratch
-  PTMP=/lustre/f2/scratch
+  STMP=/gpfs/f5/epic/scratch
```
Comment (DeniseWorthen) [inline tests/rt.sh:331]: How is someone outside of the epic group supposed to run the RTs?

Comment (ulmononian) [reply tests/rt.sh:331]: @jkbk2004 can you comment on this? i thought we switched this...

Comment (jkbk2004) [reply tests/rt.sh:331]: f5 scratch file system is maintained project-base. Users need to switch over to their own scratch project: e.g. /gpfs/f5/ncep/scratch

**Comment by RatkoVasic-NOAA** [comment]
Just to inform anyone who is not in the loop. On 2/6/24 GFDL admins made change of permissions for the /ncrc/proj/epic directory WITHOUT prior notice.
Ticket is opened with GFDL helpdesk. For your reference and inquires with GFDL helpdesk ticket number is [GFDL#5023877].

**Comment by jkbk2004** [comment]
#2124 was combined to this pr. @BrianCurtis-NOAA This PR is ready.

**Comment by jkbk2004** [comment]
@zach1221 Hercules test log is at /work/noaa/epic/jongkim/UFS-RT/hercules/RegressionTests_hercules.log. I started seeing gnv1_nested_intel comparison failure. Same symptom. Job ran ok but only issue with comparison.  We need to trun the case off on hercules. Can you follow up? @FernandoAndrade-NOAA can you check jet log? if we continue to see a case or two for time-out issue, we need to turn them off. Jet resources are not that affordable.

**Comment by FernandoAndrade-NOAA** [comment]
> @zach1221 Hercules test log is at /work/noaa/epic/jongkim/UFS-RT/hercules/RegressionTests_hercules.log. I started seeing gnv1_nested_intel comparison failure. Same symptom. Job ran ok but only issue with comparison. We need to trun the case off on hercules. Can you follow up? @FernandoAndrade-NOAA can you check jet log? if we continue to see a case or two for time-out issue, we need to turn them off. Jet resources are not that affordable.

Just leaving a note that the jet log is missing the cpld_debug_gfsv17 test due to a timeout. Jet seems to have an issue recently generating complete RT logs with that test as well as not posting the summary of results following a failure. It looks like the hash info didn't generate either at the beginning. I'm currently rerunning on jet with an increased time limit, I'll append the log when it finishes.

**Comment by FernandoAndrade-NOAA** [comment]
> @zach1221 Hercules test log is at /work/noaa/epic/jongkim/UFS-RT/hercules/RegressionTests_hercules.log. I started seeing gnv1_nested_intel comparison failure. Same symptom. Job ran ok but only issue with comparison. We need to trun the case off on hercules. Can you follow up? @FernandoAndrade-NOAA can you check jet log? if we continue to see a case or two for time-out issue, we need to turn them off. Jet resources are not that affordable.

I've disabled cpld_debug_gfsv17 in rt.conf for jet. This test has consistently failed on my end due to timeouts when running RTs on jet.

**Comment by zach1221** [comment]
Testing is complete. We can move on to review and merge process.

**Comment by zach1221** [comment]
@ulmononian fv3atm sub-pr #778 has been merged. Can you please update the submodule hash and revert .gitmodule url?
https://github.com/NOAA-EMC/fv3atm/commit/28bfc365eb65364f1b4972a870837640cb7331a3

**Review by jkbk2004 — APPROVED** [review]

**Review by zach1221 — APPROVED** [review]

**Pull request merged**